tools/rosgraph/src/rosgraph/roslogging.py

```diff
@@ -174,33 +174,28 @@ def __init__(self, colorize=True):
     def emit(self, record):
         level, color = _logging_to_rospy_names[record.levelname]
         record_message = _defaultFormatter.format(record)
-        if 'ROSCONSOLE_FORMAT' in os.environ.keys():
-            msg = os.environ['ROSCONSOLE_FORMAT']
-            msg = msg.replace('${severity}', level)
-            msg = msg.replace('${message}', str(record_message))
-            msg = msg.replace('${walltime}', '%f' % time.time())
-            msg = msg.replace('${thread}', str(record.thread))
-            msg = msg.replace('${logger}', str(record.name))
-            msg = msg.replace('${file}', str(record.pathname))
-            msg = msg.replace('${line}', str(record.lineno))
-            msg = msg.replace('${function}', str(record.funcName))
-            try:
-                from rospy import get_name
-                node_name = get_name()
-            except ImportError:
-                node_name = '<unknown_node_name>'
-            msg = msg.replace('${node}', node_name)
-            if self._get_time is not None and not self._is_wallclock():
-                t = self._get_time()
-            else:
-                t = time.time()
-            msg = msg.replace('${time}', '%f' % t)
-            msg += '\n'
+        msg = os.environ.get('ROSCONSOLE_FORMAT',
+                             '[${severity}] [${time}]: ${message}')
+        msg = msg.replace('${severity}', level)
+        msg = msg.replace('${message}', str(record_message))
+        msg = msg.replace('${walltime}', '%f' % time.time())
+        msg = msg.replace('${thread}', str(record.thread))
+        msg = msg.replace('${logger}', str(record.name))
+        msg = msg.replace('${file}', str(record.pathname))
+        msg = msg.replace('${line}', str(record.lineno))
+        msg = msg.replace('${function}', str(record.funcName))
+        try:
+            from rospy import get_name
+            node_name = get_name()
+        except ImportError:
+            node_name = '<unknown_node_name>'
+        msg = msg.replace('${node}', node_name)
+        if self._get_time is not None and not self._is_wallclock():
+            t = self._get_time()
         else:
-            msg = '[%s] [WallTime: %f]' % (level, time.time())
-            if self._get_time is not None and not self._is_wallclock():
-                msg += ' [%f]' % self._get_time()
-            msg += ' %s\n' % record_message
+            t = time.time()
+        msg = msg.replace('${time}', '%f' % t)
```

Comment:
This is relevant C++ code for the non-trivial expansion of `${time}`:

https://github.com/ros/ros_comm/blob/kinetic-devel/tools/rosconsole/src/rosconsole/rosconsole.cpp#L206-L213